python_scripts/trees_sol_01.py

```diff
@@ -150,46 +150,38 @@
 
 # %% tags=["solution"]
 import numpy as np
```

Comment:
I think that we can simplify the code by using some `groupby` that will avoid using the `loc` here:

```python
from matplotlib import cm

_, axs = plt.subplots(ncols=3, nrows=1, sharey=True, figsize=(12, 5))
plt.suptitle("Predicted probabilities for decision tree model", y=1.05)
plt.subplots_adjust(bottom=0.45)

for idx, (class_of_interest, data_class) in enumerate(
    data_test.groupby(by=target_test)
):
    axs[idx].set_title(f"Class {class_of_interest}")
    disp = DecisionBoundaryDisplay.from_estimator(
        tree,
        data_test,
        response_method="predict_proba",
        class_of_interest=class_of_interest,
        ax=axs[idx],
        vmin=0,
        vmax=1,
    )
    data_class.plot.scatter(
        x="Culmen Length (mm)",
        y="Culmen Depth (mm)",
        ax=axs[idx],
        marker="o",
        s=100,
        c="w",
        edgecolor="k",
    )

ax = plt.axes([0.15, 0.14, 0.7, 0.05])
plt.colorbar(
    cm.ScalarMappable(cmap="viridis"), cax=ax, orientation="horizontal"
)
_ = plt.title("Probability")
```

Reply:
I feel that we have less matplotlib boilerplate code in this case since we rely on pandas directly.